Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -2443,11 +2391,11 @@ class CombineObjectServerTests: SwiftSyncTestCase {
         let objIds = collection.insertMany([document, document2, document3, document4]).await(self)
         let objectIds = objIds.map { $0.objectIdValue! }
 
-        var watchEx1 = expectation(description: "Main thread watch")
-        var watchEx2 = expectation(description: "Background thread watch")
+        let watchEx1 = Locked(expectation(description: "Main thread watch"))
```

Comment:
Why do we need a lock to access this expectations?

Reply:
Combine and sendable checking don't really combine well. The main thread one will always be used on the main thread, but the compiler doesn't know that. It could safely just be `Unchecked` instead of `Locked`, but I'm inclined to say that we should always be using `Locked` in tests when we need to silence a sendability warning even if it shouldn't be required simply because it's less likely to result in weird mysterious problems if we're wrong about it not being required. In non-test code the perf overhead is more of a concern.

The background thread expectation actually does require locking as it's written to on the main thread and read from a dispatch queue.